### DISCUSSION TITLE
How to fix broken install on Amazon Linux 2022?

### DISCUSSION THREAD

**Discussion by hipwebdesign** [fail2ban/fail2ban#3374] [Q&A] [answered]
fail2ban isn't in the repo for AL2022, so I tried installing from source.

I keep getting an `AttributeError: 'dict' object has no attribute 'iteritems'` error, which I understand has to do with a change in Python3.

Following the documentation, I've tried running `./fail2ban-2to3` using the Python "2 to 3" script, but I receive an `UnicodeDecodeError: 'utf-8' codec can't decode byte 0xb7 in position 18: invalid start byte` error.

Where do I go from here?

**Comment by sebres** [comment, marked as answer]
See https://github.com/fail2ban/fail2ban/issues/2712#issuecomment-1028900927

You need a clean copy to do 2to3 (previous attempts to install created binary files/links in working copy).
